pytorch_lightning/trainer/trainer.py

```diff
@@ -1404,10 +1404,23 @@ def _call_setup_hook(self) -> None:
 
     def _call_configure_sharded_model(self) -> None:
         with self.accelerator.model_sharded_context():
-            materialize_module(self.lightning_module)
+            self._handle_meta_model()
             self.call_hook("configure_sharded_model")
             self.call_hook("on_configure_sharded_model")
 
+    def _handle_meta_model(self) -> None:
+        param = next(self.lightning_module.parameters())
+        if param.device.type != "meta":
+            return
+
+        if isinstance(self.training_type_plugin, DDPSpawnPlugin):
+            raise MisconfigurationException("LightningModule on meta device isn't supported with spawn.")
```

Comment:
does it work with TPUSpawn?

Reply:
It is a subclass of DDPSpawn, so it should fail too

Reply:
No, it doesn't. Updated !

Reply:
oh no! the update isn't required 😅 since `TPUSpawn` subclasses `DDPSpawn`. I asked to make sure to avoid this in-case it works with `TPUSpawn`.